### PR TITLE
Enable misspell, nestif golang linter

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

This PR enables misspell and nestif golang linters.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Review by mellon-collie — CHANGES_REQUESTED** [review]
Please rebase the commits to show logical progression, i.e., one commit could be to just enable go listing for `misspell` and the other commit could be for actually making the listing changes.

**Comment by mellon-collie** [comment]
Should we also raise separate PRs for enabling each of these linters and fixing them? For instance, one PR could be just for enabling `misspell` and fixing all the `misspell` lint issues.

**Comment by ankitjain235** [comment]
@mellon-collie I kept them this way so that we can track which changes were made for which enabled linter respectfully. It would be easy to track things for someone who is making changes in the same code path later.
When the PR is merged, then also the git blame would show this as part of linter enable PRs only and won't make users worry that this commit could be causing change in code behaviour. WDYT?

**Comment by mellon-collie** [comment]
> @mellon-collie I kept them this way so that we can track which changes were made for which enabled linter respectfully. It would be easy to track things for someone who is making changes in the same code path later.
> When the PR is merged, then also the git blame would show this as part of linter enable PRs only and won't make users worry that this commit could be causing change in code behaviour. WDYT?

Sure, that makes sense. Thanks for clarifying!

File: .golangci.yml [hunk 1]
```diff
@@ -22,6 +22,9 @@ linters:
     - dupl
     - exportloopref
     - importas
+    # - stylecheck It needs a lot of changes
```
Comment (PrasadG193) [inline .golangci.yml:25]: Should we remove this?

File: .golangci.yml [hunk 2]
```diff
@@ -37,6 +40,8 @@ issues:
         - unparam                       # Tests might have unused function parameters.
         - lll
         - dupl
+        - misspell
```
Comment (PrasadG193) [inline .golangci.yml:43]: I think it is fine to enable `misspell` in test file if doesn't require much changes

File: .golangci.yml [hunk 3]
```diff
@@ -60,3 +65,5 @@ linters-settings:
         alias: metav1
       - pkg: github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1
         alias: crv1alpha1
+  nestif:
+    min-complexity: 6
```
Comment (PrasadG193) [inline .golangci.yml:69]: Just curious, what is the ideal value?

File: pkg/function/restore_rds_snapshot.go [hunk 4]
```diff
@@ -199,11 +199,7 @@ func restoreRDSSnapshot(
 		// If securityGroupID arg is nil, we will try to find the sgIDs by describing the existing instance
 		// Find security group ids
 		if sgIDs == nil {
-			if !isAuroraCluster(string(dbEngine)) {
-				sgIDs, err = findSecurityGroups(ctx, rdsCli, instanceID)
-			} else {
-				sgIDs, err = findAuroraSecurityGroups(ctx, rdsCli, instanceID)
-			}
+			sgIDs, err = findSecurityGroupIDs(ctx, rdsCli, instanceID, string(dbEngine))
```
Comment (PrasadG193) [inline pkg/function/restore_rds_snapshot.go:202]: same here

File: pkg/function/create_rds_snapshot.go [hunk 5]
```diff
@@ -160,6 +139,36 @@ func createRDSSnapshot(ctx context.Context, instanceID string, dbEngine RDSDBEng
 	return output, nil
 }
 
+func createSnapshot(ctx context.Context, rdsCli *rds.RDS, snapshotID, dbEngine, instanceID string) (int64, error) {
```
Comment (PrasadG193) [inline pkg/function/create_rds_snapshot.go:142]:
```suggestion
func createRDSSnapshot(ctx context.Context, rdsCli *rds.RDS, snapshotID, dbEngine, instanceID string) (int64, error) {
```

File: .golangci.yml (hunk 2 above)
Comment (ankitjain235) [reply .golangci.yml:42]: It does require changes. Enabling these linters for tests would need a lot of extra work. I'll create a separate issue for the same?

**Review by PrasadG193 — APPROVED** [review]

**Pull request merged**